server/src/main/java/org/elasticsearch/script/Metadata.java

```diff
@@ -55,26 +55,50 @@ public class Metadata {
     public static FieldProperty<Number> LongField = new FieldProperty<>(Number.class).withValidation(FieldProperty.LONGABLE_NUMBER);
 
     protected final Map<String, Object> map;
-    protected final Map<String, FieldProperty<?>> properties;
+    private final Map<String, FieldProperty<?>> properties;
     protected static final FieldProperty<?> BAD_KEY = new FieldProperty<>(null, false, false, null);
 
-    public Metadata(Map<String, Object> map, Map<String, FieldProperty<?>> properties) {
+    /**
+     * Constructs a new Metadata object represented by the given map and properties.
+     * <p>
+     * The passed-in map is used directly -- subsequent modifications to it outside the methods of this class may result in
```

Comment:
Also, the map is expected to be modifiable.

Reply:
👍, e744f5c